src/resultset/formatters/resultset_replyverbose.c

```diff
@@ -89,14 +89,15 @@ static void _ResultSet_VerboseReplyWithNode(RedisModuleCtx *ctx, GraphContext *g
 	// ["labels", [label (string)]]
 	RedisModule_ReplyWithArray(ctx, 2);
 	RedisModule_ReplyWithStringBuffer(ctx, "labels", 6);
-	// Print label in nested array for multi-label support
+	const char *label = NODE_GET_LABEL(n);
 	// Retrieve label if it is not set on the node.
 	// TODO Make a more efficient lookup for this string
-	const char *label = n->label ? n->label : GraphContext_GetNodeLabel(gc, n);
+	if(label == NULL) label = GraphContext_GetNodeLabel(gc, n);
```

Comment:
If the sequence of retrieving node label always looks like this:

1. `NODE_GET_LABEL(n)`
2. `GraphContext_GetNodeLabel(gc, n)`

Consider modifying `NODE_GET_LABEL` to call `GraphContext_GetNodeLabel` if the local node label isn't set.